### PR TITLE
Fix rare count_test edge case (#2)

### PR DESCRIPTION
**Pull request by maksymiuks**
Hi!

In our enterprise solution, we have discovered a peculiar edge case when `count_test` function is called with `.current_test` not yet initialized, leading to error:

```
Error in tests$.data[[1L]] <- .current_test$index : 
  replacement has length zero
```

For us, we encountered it when calculating coverage for the  `digest` when using `tinytest` Ubuntu libraries (funnily enough, when installing tinytest from source, the issue did not appear). That makes reproducing the issue really tedious, although I was able to do it on macOS as well. Nevertheless, I believe that there is no harm in adding that if statement to cover that edge case, even if it's super rare, simply to make sure the `covr` is not vulnerable to weird/unexpected package combinations in the future.

### CODE REVIEW

File: R/trace_tests.R [hunk 1]
```diff
@@ -114,6 +114,9 @@ count_test <- function(key) {
     tests$tally <- rbind(tests$tally, matrix(NA_integer_, ncol = 4L, nrow = n))
   }
 
+  # ignore if .current_test was not initialized properly yet
+  if (length(.current_test$index) == 0) return()
```
Comment (jimhester) [inline R/trace_tests.R]:
```suggestion
  if (length(.current_test$index) == 0) {
    return()
  }
```

**Review by jimhester — COMMENTED** [review]
Had one minor style change suggestion.

Also can you please add a bullet to NEWS? It should briefly [describe the change](https://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

**Comment by maksymiuks** [comment]
@jimhester, I updated the PR to include another edge-case fix. I also added an issue for both these problems so they can be linked in NEWS. Could you please take a look at it?

**Comment by jimhester** [comment]
Sounds good, thanks for working on this!
